validator/src/main.rs

```diff
@@ -168,21 +140,48 @@ fn start_gossip_node(
 
 fn get_rpc_node(
     cluster_info: &ClusterInfo,
+    entrypoint_gossip: &SocketAddr,
     validator_config: &ValidatorConfig,
     blacklisted_rpc_nodes: &mut HashSet<Pubkey>,
     snapshot_not_required: bool,
     no_untrusted_rpc: bool,
 ) -> (ContactInfo, Option<(Slot, Hash)>) {
     let mut blacklist_timeout = Instant::now();
     loop {
-        info!(
-            "Searching for an RPC service, shred version={:?}...",
-            validator_config.expected_shred_version
-        );
         sleep(Duration::from_secs(1));
         info!("\n{}", cluster_info.contact_info_trace());
 
-        let rpc_peers = get_shred_rpc_peers(&cluster_info, validator_config.expected_shred_version);
+        let shred_version = validator_config
+            .expected_shred_version
+            .unwrap_or_else(|| cluster_info.my_shred_version());
+        if shred_version == 0 {
+            if let Some(entrypoint) =
+                cluster_info.lookup_contact_info_by_gossip_addr(entrypoint_gossip)
+            {
+                if entrypoint.shred_version == 0 {
```

Comment:
If we're 0 and the entrypoint is also 0 (ie, a `solana-gossip spy` without a `--shred-version`), we need user input to proceed.  Abort the process